### DISCUSSION TITLE
zigbee-herdsman-converters new device

### DISCUSSION THREAD

**Discussion by Tomlox2020** [Koenkk/zigbee2mqtt#24517] [General]
Please excuse the question, but somehow I don't understand how to integrate a new component using the herdsman converter. I have read the instructions several times, but I don't have the option “Generate external definition” in the DEV console, for example. 
I would like to integrate the Zigbee JETSTRÖM L2205. This is not yet stored in the Zigbee2Mqtt plugin for the well-known Ikea Zigbee lamps.

I have a Raspberry 4 running the Zigbee2Mqtt. I am using a Sonoff Dongle ZigBee 3.0 as an adapter.

Could someone briefly describe how to integrate this lamp using the converter?

**Comment by Tomlox2020** [comment]
@Koenkk  can you help me please?

**Comment by clockbrain** [comment]
Have you paired the device? If so it should appear in the front end devices list - likely labelled with its MAC address.
Click on that device name, click on the Dev Console tab, and the 'generate_external_definition' button should be right at the top.
<img width="698" alt="image" src="https://github.com/user-attachments/assets/3f762e02-e1e4-43a2-a4fc-158addd7679a">

**Reply by Tomlox2020** [reply]
Thanks, I'll test it right away this evening.

**Comment by Tomlox2020** [comment]
Unfortunately, I do not see this selection in my menu.
![IMG_0016](https://github.com/user-attachments/assets/003731f4-e876-4ae4-86fa-48be267beec1)
![IMG_0015](https://github.com/user-attachments/assets/bac3302c-6495-4745-b816-f73601154b66)
![IMG_0014](https://github.com/user-attachments/assets/8512d926-ac2e-4d05-a9ee-7fa0bc20ee48)

**Comment by clockbrain** [comment]
I don't know why the 'generate_external_definition' isn't there. Maybe something related to the translation?

**Reply by Tomlox2020** [reply]
Yes thats true, when i open the „Developer Console“ i only see this. (Picture)
![IMG_3782](https://github.com/user-attachments/assets/9f217ab8-9211-4d43-83e8-64a4a3307ef9)

**Reply by clockbrain** [reply]
What version of zigbee2mqtt are you running?

**Reply by Tomlox2020** [reply]
Version 2.3.0
![IMG_3783](https://github.com/user-attachments/assets/deae78d7-f635-4b7f-983a-0e33c5e44757)

**Comment by clockbrain** [comment]
Looks like your version is too old.

Here it states that Loxberry Release 2.3.0 has z2m version 1.32.2
https://wiki.loxberry.de/plugins/zigbee2mqtt_plugin/start?do=export_pdf

Here it states that z2m version 1.35.0 was where generating external definitions was added
https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.35.0

So I think you need to be on at least Release 2.5.0 of Loxberry. A more recent version would be even better.

**Reply by Tomlox2020** [reply]
Aaaahhh o.k. many thanks, i will update my system tomorrow and check it.

**Comment by Tomlox2020** [comment]
sorry for the late feedback, unfortunately I didn't have time to update.

I have now done this and now the plugin does not work at all. every time I go to the “zigbee2mqtt UI” tab I get the following screen and no devices are displayed.

I have now installed the following versions:
Loxberry 3.0.1.2
zigbee2mqtt 2.10.0
![Unbenannt](https://github.com/user-attachments/assets/612f24a2-f810-4730-89c7-f60ea7299809)

**Reply by Tomlox2020** [reply]
here is the log file
![image](https://github.com/user-attachments/assets/700c3413-0a64-442f-b095-e67048302451)

**Reply by clockbrain** [reply]
You said in your very first post that you have a Sonoff Dongle ZigBee 3.0 adapter. The log above shows it using the default zstack driver. Are you sure you have the following in your configuration.yaml?
```
serial:
    adapter: ember
```
Or perhaps its not finding or loading your configuration.yaml.

**Reply by Tomlox2020** [reply]
This is the adapter
![Adapter](https://github.com/user-attachments/assets/deae5f80-bbdc-4d75-93b7-ccd03bbb8e75)

**Reply by Tomlox2020** [reply]
![image](https://github.com/user-attachments/assets/2c73846d-bf0e-43a7-8e76-601514b42b76)

**Reply by clockbrain** [reply]
My mistake. Yours is the ZBdongle-P which is indeed the zstack driver.

Mine is the ZBdongle-E which uses the Ember driver. They both have similar /dev/serial/by-id/ paths which confused me. Only difference is Dongle_Plus_V2 vs Dongle_Plus.

I suggest you use the /dev/serial/by-id/usb-ITEAD_SONOFF_Zigbee_3.0_USB_Dongle_Plus_XXXXXXXXXXXXXXX path in your configuration.yaml as that will be more reliable than the /dev/ttyUSB0. This may 

Also, don't post your network key. That is a security feature.

From the log file it suggests a clash of panId or extended panId with another network. If this is a second test network make sure to give it a different network key. Set these to GENERATE in your configuration.yaml to generate a new random value.
https://www.zigbee2mqtt.io/guide/configuration/zigbee-network.html#network-config

**Reply by Tomlox2020** [reply]
Hello, thanks for the tips, I have done all of them. Unfortunately it still doesn't work, I don't understand it.
![new_1](https://github.com/user-attachments/assets/637617cf-6ffe-4f9b-8b79-650019f5c180)

**Comment by Tomlox2020** [comment]
Hey, now i have configured mqtt and it works😊
Thanks a lot for your Help😊

**Comment by Tomlox2020** [comment]
@clockbrain 
Can I transfer the learned devices to another Raspy with the zigbee2mqtt plugin installed or do I have to learn them all again there?

**Reply by clockbrain** [reply]
@Tomlox2020 that depends... See https://www.zigbee2mqtt.io/guide/faq/#how-do-i-move-my-zigbee2mqtt-instance-to-a-different-environment
TLDR - if you move your coordinator stick and your z2m/data folder over you should not have to join your devices again.
